src/main/java/bg/codexio/springframework/data/jpa/requery/resolver/FilterJsonArgumentResolver.java

```diff
@@ -82,26 +84,22 @@ public Object resolveArgument(
             @NotNull NativeWebRequest webRequest,
             WebDataBinderFactory binderFactory
     ) throws Exception {
-        var filterJson = webRequest.getParameter("filter");
-        var complexFilterJson = webRequest.getParameter("complexFilter");
+        HttpServletRequest request = webRequest.getNativeRequest(HttpServletRequest.class);
+        FilterRequestWrapper filterWrapper = this.httpFilterAdapter.adapt(request);
 
         var genericType =
                 (Class<?>) ((ParameterizedType) parameter.getGenericParameterType()).getActualTypeArguments()[0];
 
-        if (filterJson != null) {
-            return this.getSimpleFilterSpecification(
-                    filterJson,
-                    genericType
-            );
-        }
+        Specification<Object> specification = this.noFilterSpecification();
+        filterWrapper.ifSimple(simpleFilter -> {
```

Comment:
Can we make it returnable? So you can just do:

```
return wrapper.ifSimple(...).orComplex(...).or(noFilter);
```

Reply:
Probably the `or` (`else`, whatever you name it) method should be finalizer (the others may still return `this`).

Reply:
done